### ISSUE TITLE
Bug ? How to disable interaction on frontview when rearview is revealed

### ISSUE THREAD

**Issue by BaranBerk** [John-Lluch/SWRevealViewController#410]
When Menu opened user can interact with the other part but menu should closed??

**Comment by ridvank** [comment]
You can use SWRevealViewControllerDelegate;
    func revealController(revealController: SWRevealViewController!, didMoveToPosition position: FrontViewPosition) {
        if(position == FrontViewPosition.Left){
            self.view.userInteractionEnabled = true
        }
        else {
            self.view.userInteractionEnabled = false
        }
    }

**Comment by ElectroBuddha** [comment]
Thanks ridvank, this was Exactly what I needed!

**Comment by Solublepeter** [comment]
Note that this does prevent swiping back to the front view controller (although the navigation button does still work) as all user interaction is disabled.

Someone has posted a solution to this as one of the other answers.

**Comment by ridvank** [comment]
Nope it does not prevent swipe. I wanted to disable user interaction on the view which make sense. If you add Gesture recognizer it does not prevent to swipe.

**Comment by Solublepeter** [comment]
I have a UIViewController, containing a UITableView, it has the recommended code to add a gesture recogniser and set itself as a delegate in ViewDidLoad:

        revealViewController.delegate=self;
        [self.sidebarButton setTarget: self.revealViewController];
        [self.sidebarButton setAction: @selector( revealToggle: )];
        [self.view addGestureRecognizer:self.revealViewController.panGestureRecognizer];
    
If I add the two routines for willMoveToPosition and didMoveToPosition to this ViewController, the swipe gesture no longer works on this ViewController. (simply remming out your two routines makes it work again)

Am I missing something?

**Comment by Solublepeter** [comment]
I think I got it...

If you are using a UIViewController containing a UITableView, instead of a UITableViewController, you need to change each instance of "self.view.interactionEnabled" in those new routines to "self.tableViewName.interactionEnabled", then it all seems to work fine.

**Comment by ridvank** [comment]
You are totally right. I have a UICollectionView in my ViewController. so i disabled CollectionView interaction.

**Comment by BaranBerk** [comment]
On your MenuViewController:
-(void)viewWillAppear:(BOOL)animated {
         [super viewWillAppear:animated];
         [self.revealViewController.frontViewController.view setUserInteractionEnabled:NO];
        [self.revealViewController.view         addGestureRecognizer:self.revealViewController.panGestureRecognizer];
}  
-(void)viewDidDisappear:(BOOL)animated {
      [super viewDidDisappear:animated];
      [self.revealViewController.frontViewController.view setUserInteractionEnabled:YES];
}

And on your menu items’ view controller (you can make a ParentViewController for all of them):
-(void)viewDidLoad {
     [super viewDidLoad];
    SWRevealViewController *revealController = [self revealViewController];
    [revealController panGestureRecognizer];
    [revealController tapGestureRecognizer];
}

I find it in this site many thanks to them:
http://www.tech.theplayhub.com/swrevealviewcontroller_remove_interaction_on_frontview_when_rearview_is_revealed/

**Comment by cbh2000** [comment]
Duplicate of #321.
